src/parser/json/creation_info_parser.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2022 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datetime import datetime
+from typing import Dict, Optional, List
+
+from src.model.actor import Actor
+from src.model.checksum import Checksum
+from src.model.document import CreationInfo
+from src.model.external_document_ref import ExternalDocumentRef
+from src.model.version import Version
+from src.parser.error import SPDXParsingError
+from src.parser.json.actor_parser import ActorParser
+from src.parser.json.checksum_parser import ChecksumParser
+from src.parser.json.dict_parsing_functions import append_parsed_field_or_log_error, datetime_from_str, \
+    raise_parsing_error_if_logger_has_messages, construct_or_raise_parsing_error, parse_field_or_log_error
+from src.parser.logger import Logger
+
+
+class CreationInfoParser:
+    logger: Logger
+    actor_parser: ActorParser
+    checksum_parser: ChecksumParser
+
+    def __init__(self):
+        self.logger = Logger()
+        self.actor_parser = ActorParser()
+        self.checksum_parser = ChecksumParser()
+
+    def parse_creation_info(self, doc_dict: Dict) -> CreationInfo:
+        logger = Logger()
+        spdx_version: str = doc_dict.get("spdxVersion")
+        spdx_id: str = doc_dict.get("SPDXID")
+        name: str = doc_dict.get("name")
+        document_namespace: str = doc_dict.get("documentNamespace")
+        creation_info_dict: Dict = doc_dict.get("creationInfo")
+
+        # There are nested required properties. If creationInfo is not set, we cannot continue parsing.
+        if creation_info_dict is None:
+            logger.append("CreationInfo does not exist.")
+            raise SPDXParsingError([f"Error while parsing doc {name}: {logger.get_messages()}"])
+
+        list_of_creators: List[str] = creation_info_dict.get("creators")
+        creators: List[Actor] = parse_field_or_log_error(logger=logger, field=list_of_creators,
```

Comment:
This will throw a TypeError when `creators` is `None`

Reply:
fixed with the changes in `parse_field_or_log_error`, but I also added a testcase for this